sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -312,6 +313,24 @@ case class InputAdapter(child: SparkPlan) extends UnaryExecNode with CodegenSupp
 object WholeStageCodegenExec {
   val PIPELINE_DURATION_METRIC = "duration"
 
+  private val codegenStageCounter = ThreadLocal.withInitial(new Supplier[Integer] {
```

Comment:
Instead of adding these logics into `object WholeStageCodegenExec`, could we just create a new object?

Reply:
But IMHO this is the place to put it, since it's closely tied to the initialization of `WholeStageCodegenExec` object instances.
Are you suggesting something like a `object WholeStageCodegenId` and move the newly added methods there instead? That would work too.

Reply:
Yeah. Creating a new `object WholeStageCodegenId` is better.